rcl/src/rcl/time.c

```diff
@@ -453,19 +454,18 @@ rcl_clock_remove_jump_callback(
   }
 
   // Shrink size of the callback array
-  if (clock->num_jump_callbacks == 1) {
+  if (clock->num_jump_callbacks-- == 1) {
```

Comment:
I think this change may need to go into a separate PR, based on the issue below.

Reply:
It could go on a separate PR, but it goes hand in hand with the test that prevents a regression.

Reply:
Is this fixing a bug and should it be backported, or is it just some cleanup?

Reply:
Hmm, that is a good point. I think it could be backported, yeah. It adds safety checks and ensure `rcl_remove_jump_callback()` fails a reallocation safely i.e. that you can continue despite a failed a reallocation.